src/components/commonModal/CommonModal.js

```diff
@@ -0,0 +1,23 @@
+// children to komponent który chcemy wyświetlić w modalu
+// callback to funkcja która narazie jest ustawiona na sztywno, która po kliknięciu na modal ma za zadanie wyłączyć go.
+// DOCELOWO FUNKCJĄ TĄ MA BYĆ "hideElement" zdefiniowana w utils => functions
+
+export const CommonModal = (children, callback)=>{
+
+  const modal = document.createElement('div');
+  const app = document.getElementById('app-structure');
+  modal.classList.add('modal');
+  modal.setAttribute('data-testid', 'html-modal');
+  app.classList.add('modalEffect');
+  let childrenHtmlTemplate = `<div>${children}</div>`;
+  modal.innerHTML = childrenHtmlTemplate;
+  modal.addEventListener('click', ()=>{
+    callback(modal);
+    removeModalEffect(app, 'modalEffect');
+  });
+  return modal
+};
```

Comment:
```suggestion
export const ModalWindow = (children, callback)=>{

  const component = document.createElement('div');
  const app = document.getElementById('app-structure');
  component.classList.add('modal');
  component.setAttribute('data-testid', 'html-modal');
  app.classList.add('modalEffect');
  const childrenHtmlTemplate = `<div>${children}</div>`;
  component.innerHTML = childrenHtmlTemplate;
  component.addEventListener('click', ()=>{
    callback(component);
    removeModalEffect(app, 'modalEffect');
  });

  const modal = {
    show() {
      return component;
    },
    hide() {
      component.style.display = 'none'
      return component;
    }
  }
  return modal
};
```
Co myślisz o tym, żeby nasz komponent był samowystarczalny i mógł znikać po wywołaniu jego funkcji hide?

Reply:
Poprawiłam za Twoją sugestią. Czy callback jest teraz konieczny?